### PR TITLE
tests: make input plugin testing run more stable on Travis CI

### PR DESCRIPTION
**Pull request by fujimotos**
Previously we used a few clever tricks in order to make our test
cases run robustly on slow servers (e.g. use while loop to absorb
random delays). Still we saw occasional failures on CI servers.

My opinion is that the old implementation was thinking too complex,
ended up introducing another fragility itself. I do not think adding
another pile of tricks would improve the overall situation.

Instead, this patch simplifies the test code significantly (e.g.
use a fixed sleep interval instead of a wait loop), and improve
the stability by making sure each test has enough cushion for
absorbing random delays.

In a nutshell, the new version tests the event generation using the
following schedule:

    0  1  2  3  4  5  6  7  8 (sec)
    |--C--*--F--C--*--F--C--|

    F ... Flush (3 sec interval)
    C ... Check if any data has been generated

... and should ensure that each input plugin generates events in a
timely manner.

Issue Link: #1231

### CODE REVIEW

**Comment by edsiper** [comment]
thanks @fujimotos 

Not associated to the root cause of the problem but since this is a test refactor: would you please make some adjustments/tests decreasing the Flush interval ?, actually the Flush option supports the value SECONDS or SECONDS.NANOSECONDS, so you can do something like Flush 0.500000000 (for 0.5 seconds). 

note: I am strongly considering to move away from Travis CI due to several reasons, candidate is Gitlab CI

**Comment by fujimotos** [comment]
@edsiper Thanks! That's nice to hear about Gitlab CI.

I've updated `in_simple_systems.c` to use 0.5 sec interval. Now it
follows this schedule:


    0     1     2     3     4   (sec)
    |--F--F--F--C--F--F--F--C--|

    F ... Flush (0.5 sec interval)
    C ... Condition checks

Probably we can shorten the check interval (it's 2s to absorb delays),
once we've migrated to a more stable CI infra.

Note: AppVayor's build failure is due to the merge of streap-processor.
I'll check that out next, and make sure the master HEAD works on MSVC.

https://ci.appveyor.com/project/fluent/fluent-bit-2e87g/build/job/2jem7a48ipm4ynj1#L388

**Comment by cosmo0920** [comment]
W/o stream processor, https://github.com/fluent/fluent-bit/pull/1236 will fix CI breakage.

**Comment by edsiper** [comment]
#1236 merged.

**Comment by fujimotos** [comment]
Rebased to the master HEAD to rerun tests.
